Dockerfile

```diff
@@ -0,0 +1,35 @@
+ARG TARGET
+
+FROM node:18 as builder
+
+# Create app directory
+WORKDIR /build
+
+# A wildcard is used to ensure both package.json AND package-lock.json are copied
+COPY package.json ./
+
+# Install app dependencies
+RUN npm install --force
+
+# Bundle app source
+COPY apps/backend/ /build/apps/backend/
+COPY tsconfig.base.json /build/tsconfig.base.json
+
+RUN ls -la
+
+# Creates a "dist" folder with the production build
+RUN npx nx build backend
+
+
+FROM node:18-alpine
+WORKDIR /app
+
+# if Target is backend then copy the backend files 
+
+COPY --from=builder /build/dist/apps/backend ./dist
```

Comment:
Building the app is only useful for production but shouldn't be done in development.